lib/Parser/Parse.cpp

```diff
@@ -1680,56 +1680,73 @@ ParseNodePtr Parser::ReferenceSpecialName(IdentPtr pid, charcount_t ichMin, char
     return pnode;
 }
 
+ParseNodePtr Parser::CreateSpecialVarDeclIfNeeded(ParseNodePtr pnodeFnc, IdentPtr pid, bool forceCreate)
+{
+    Assert(pid != nullptr);
+
+    PidRefStack* ref = pid->GetTopRef();
+
+    // If the function has a reference to pid or we set forceCreate, make a special var decl
+    if (forceCreate || (ref && ref->GetScopeId() >= m_currentBlockInfo->pnodeBlock->sxBlock.blockId))
+    {
+        return this->CreateSpecialVarDeclNode(pnodeFnc, pid);
+    }
+
+    return nullptr;
+}
+
 void Parser::CreateSpecialSymbolDeclarations(ParseNodePtr pnodeFnc, bool isGlobal)
 {
+    // Lambda function cannot have any special bindings.
+    if (pnodeFnc->sxFnc.IsLambda())
+    {
+        return;
+    }
     Assert(!(isGlobal && (this->m_grfscr & fscrEval)));
+    Assert(!isGlobal || (this->m_grfscr & fscrEvalCode));
```

Comment:
We still shouldn't get here if fscrEval is set, right? Can we still assert that?

Reply:
We are asserting this, right?

Reply:
Oh, of course. I misread the diff. This is fine.